src/test/store/symbolication.test.js

```diff
@@ -0,0 +1,187 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+// @flow
+import { storeWithProfile } from '../fixtures/stores';
+import { getProfileFromTextSamples } from '../fixtures/profiles/make-profile';
+import exampleSymbolTable from '../fixtures/example-symbol-table';
+import { SymbolStore } from '../../profile-logic/symbol-store.js';
+import * as ProfileViewSelectors from '../../reducers/profile-view';
+import { resourceTypes } from '../../profile-logic/profile-data';
+import { doSymbolicateProfile } from '../../actions/receive-profile';
+import {
+  changeSelectedCallNode,
+  changeExpandedCallNodes,
+} from '../../actions/profile-view';
+import { formatTree } from '../fixtures/utils';
+import fakeIndexedDB from 'fake-indexeddb';
+import FDBKeyRange from 'fake-indexeddb/lib/FDBKeyRange';
+import { TextDecoder } from 'text-encoding';
+
+/**
+ * Symbolication happens across actions and reducers, so test this functionality in
+ * its own file.
+ */
+describe('doSymbolicateProfile', function() {
+  function init() {
+    const profile = _createUnsymbolicatedProfile();
+    const store = storeWithProfile(profile);
+
+    return {
+      profile,
+      store,
+      namesAsFuncIndexes: (names: string[]) =>
+        names.map(name => {
+          // Get the current thread for every invocation in order to make the tests
+          // easier to read.
+          const thread = getThread(store.getState());
+          const stringIndex = thread.stringTable.indexForString(name);
+          return thread.funcTable.name.indexOf(stringIndex);
+        }),
+      symbolStore: new SymbolStore('test-db', {
+        requestSymbolTable: () => Promise.resolve(exampleSymbolTable),
+      }),
+    };
+  }
+
+  const {
+    getSelectedCallNodePath,
+    getExpandedCallNodePaths,
+    getThread,
+    getCallTree,
+  } = ProfileViewSelectors.selectedThreadSelectors;
+
+  beforeAll(function() {
+    window.indexedDB = fakeIndexedDB;
+    window.IDBKeyRange = FDBKeyRange;
+    window.TextDecoder = TextDecoder;
+  });
+
+  afterAll(function() {
+    delete window.indexedDB;
+    delete window.IDBKeyRange;
+    delete window.TextDecoder;
+  });
```

Comment:
Maybe you'd want to delete the database in `afterEach` to leave a clean state, see how we do it in `symbol-store.test.js`.